### PR TITLE
PORT-7273 | DORA Metrics

### PR DESCRIPTION
**Pull request by mk-armah**
# Description
This docs provides a comprehensive guide for creating a GitHub action computes DORA Metrics for a service (repo)

## Inputs
| Name                 | Description                                                                                          | Required | Default            |
|----------------------|------------------------------------------------------------------------------------------------------|----------|--------------------|
| repository              | GitHub repository in owner/repo format eg. port-labs/self-service-actions                               | true     | -                  |
| timeframe              | Time frame in weeks to calculate metrics on                                                              | true    | -               |
| workflow              | The name of the workflows to process. Multiple workflows can be separated by a comma (,) .                                                              | true    | -               |

- Resource Path (`/create-self-service-experiences/setup-backend/github-workflow/examples/dora-metrics`)

### CODE REVIEW

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/dora-metrics.md [hunk 1]
```diff
@@ -0,0 +1,1038 @@
...
+        github_env.write(f"lead_time_for_changes_report={report}\n")
+```
+</details>
+
+6. Trigger the action from the [self-service](https://app.getport.io/self-serve) page of your Port application. Below is a boilerplate of the expected input payload
+
+<details>
+  <summary><b>Sample Input</b></summary>
+:::tip
+- `<Entity-ID>` - your target entity id in port
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/dora-metrics.md:1024]:
```suggestion
- `<Entity-ID>` - your target entity identifier in Port
```

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/dora-metrics.md [hunk 2]
```diff
@@ -0,0 +1,1038 @@
...
+- `<GITHUB-ORG>` - your GitHub organization or user name.
+- `<GITHUB-REPO-NAME>` - your GitHub repository name.
+:::
+
+```json showLineNumbers title="sample input json"
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/dora-metrics.md:1029]:
```suggestion
```json showLineNumbers
```

File: docs/create-self-service-experiences/setup-backend/github-workflow/examples/dora-metrics.md [hunk 3]
```diff
@@ -0,0 +1,1038 @@
...
+
+```text showLineNumbers title="requirements.txt"
+PyGithub==2.3.0
+requests==2.31.0
+pytz==2024.1
+```
+
+</details>
+
+5. Create the following python scripts (`calculate_pr_metrics.py`, `deployment_frequency.py` and `lead_time_for_changes.py` ) in a folder named `dora` at the root of your GitHub repository:
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/github-workflow/examples/dora-metrics.md:584]:
```suggestion
5. Create the following Python scripts (`calculate_pr_metrics.py`, `deployment_frequency.py` and `lead_time_for_changes.py` ) in a folder named `dora` at the root of your GitHub repository:
```

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Some initial comments

**Review by hadar-co — APPROVED** [review]

**Pull request merged**